crypto/bsslmlkem/mlkem768.c

```diff
@@ -23,6 +23,14 @@
 # include <stdio.h>
 #endif
 
+/*
+ * Set/unset to test performance impact of cached SHA3 EVP_MD instances
+ * Impact on x64: everything about 8% slower if un-set
+ * TODO(ML-KEM): Remove define if satisfied with approach (API impact)
+ * removal will also fix style-related problem
+ */
```

Comment:
We could maybe also gain some more performance if we, instead of keeping EVP_MDs around, kept EVP_MD_CTXes preinitialized with the appropriate EVP_MD. However it might be also a premature optimization.

BTW the code without USE_SHA3_CACHE leaks the EVP_MDs, so it definitely should not be kept.

Reply:
> BTW the code without USE_SHA3_CACHE leaks the EVP_MDs, so it definitely should not be kept.

ACK. I will delete before merge to "bsslmlkem" but wanted to give interested parties the chance to experience the performance difference (~8% better due to caching).

Reply:
so now done.